pkg/dockerregistry/server/blobdescriptorservice.go

```diff
@@ -34,61 +30,50 @@ func (b ByGeneration) Less(i, j int) bool { return b[i].Generation > b[j].Genera
 func (b ByGeneration) Len() int           { return len(b) }
 func (b ByGeneration) Swap(i, j int)      { b[i], b[j] = b[j], b[i] }
 
-func init() {
-	err := middleware.RegisterOptions(storage.BlobDescriptorServiceFactory(&blobDescriptorServiceFactory{}))
-	if err != nil {
-		logrus.Fatalf("Unable to register BlobDescriptorServiceFactory: %v", err)
-	}
-}
+type blobDescriptorServiceFactoryFunc func(svc distribution.BlobDescriptorService) distribution.BlobDescriptorService
 
-// blobDescriptorServiceFactory needs to be able to work with blobs
-// directly without using links. This allows us to ignore the distribution
-// of blobs between repositories.
-type blobDescriptorServiceFactory struct{}
-
-func (bf *blobDescriptorServiceFactory) BlobAccessController(svc distribution.BlobDescriptorService) distribution.BlobDescriptorService {
-	return &blobDescriptorService{svc}
+func (f blobDescriptorServiceFactoryFunc) BlobAccessController(svc distribution.BlobDescriptorService) distribution.BlobDescriptorService {
+	return f(svc)
 }
 
 type blobDescriptorService struct {
 	distribution.BlobDescriptorService
+	repo *repository
+}
+
+func (r *repository) BlobDescriptorService(svc distribution.BlobDescriptorService) distribution.BlobDescriptorService {
```

Comment:
Please move this function to `repository.go`

Reply:
I'd like to move as few functions as possible in this PR.